src/EventStore.Client.PersistentSubscriptions/EventStorePersistentSubscriptionsClient.List.cs

```diff
@@ -0,0 +1,82 @@
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using EventStore.Client.PersistentSubscriptions;
+
+#nullable enable
+namespace EventStore.Client {
+	partial class EventStorePersistentSubscriptionsClient {
+		/// <summary>
+		/// Lists all persistent subscriptions for the $all stream
+		/// </summary>
+		/// <param name="userCredentials"></param>
+		/// <param name="cancellationToken"></param>
+		/// <returns></returns>
+		public IAsyncEnumerable<PersistentSubscriptionDetail> ListForAllAsync(UserCredentials? userCredentials = null,
+			CancellationToken cancellationToken = default) {
+
+			var req = new ListReq() {
+				Options = new ListReq.Types.Options{
+					ListForStream = new ListReq.Types.StreamOption() {
+						All = new Empty()
+					}
+				}
+			};
+
+			return ListInternalAsync(req, userCredentials, cancellationToken);
+		}
+
+		/// <summary>
+		/// Lists all persistent subscriptions for the stream
+		/// </summary>
+		/// <param name="stream"></param>
+		/// <param name="userCredentials"></param>
+		/// <param name="cancellationToken"></param>
+		/// <returns></returns>
+		public IAsyncEnumerable<PersistentSubscriptionDetail> ListForStreamAsync(string stream, UserCredentials? userCredentials = null,
+			CancellationToken cancellationToken = default) {
+
+			var req = new ListReq() {
+				Options = new ListReq.Types.Options{
+					ListForStream = new ListReq.Types.StreamOption() {
+						Stream = stream
+					}
+				}
+			};
+			
+			return ListInternalAsync(req, userCredentials, cancellationToken);
+		}
+		
+		/// <summary>
+		/// Lists all persistent subscriptions
+		/// </summary>
+		/// <param name="userCredentials"></param>
+		/// <param name="cancellationToken"></param>
+		/// <returns></returns>
+		public IAsyncEnumerable<PersistentSubscriptionDetail> ListAllAsync(UserCredentials? userCredentials = null,
+			CancellationToken cancellationToken = default) {
+
+			var req = new ListReq() {
+				Options = new ListReq.Types.Options{
+					ListAllSubscriptions = new Empty()
+				}
+			};
+			
+			return ListInternalAsync(req, userCredentials, cancellationToken);
+		}
+		
+		private async IAsyncEnumerable<PersistentSubscriptionDetail> ListInternalAsync(ListReq req,
+			UserCredentials? userCredentials, [EnumeratorCancellation] CancellationToken cancellationToken) {
+
+			using var call = new PersistentSubscriptions.PersistentSubscriptions.PersistentSubscriptionsClient(
```

Comment:
**QUESTION:** Shouldn't we have here the HTTP fallback?

Reply:
Done.